src/OVAL/results/oval_resultSystem.c

```diff
@@ -205,14 +214,20 @@ struct oval_result_definition *oval_result_system_get_new_definition
 		char *id = oval_definition_get_id(oval_definition);
 		rslt_definition = oval_result_system_get_definition(sys, id);
 		if (rslt_definition == NULL) {
-			rslt_definition = make_result_definition_from_oval_definition(sys, oval_definition,
+			rslt_definition = make_result_definition_from_oval_definition(sys, oval_definition, visited_definitions,
 				variable_instance ? variable_instance : 1);
+			if (rslt_definition == NULL) {
+				return NULL;
+			}
```

Comment:
I would say that these braces are redundant.

Reply:
Yes, you're right, the Linux kernel coding style discourages that. I usually do this to prevent forgetting them if the single statement would be replaced by multiple statements in future. I will remove them quickly.